libs/sdk-core/src/error.rs

```diff
@@ -30,8 +47,11 @@ pub enum SdkError {
     #[error("Failed to receive payment: {err}")]
     ReceivePaymentFailed { err: String },
 
-    #[error("Failed to send payment: {err}")]
-    SendPaymentFailed { err: String },
+    #[error("Failed to send payment: {code} {err}")]
+    SendPaymentFailed {
+        code: SendPaymentFailedCode,
+        err: String,
+    },
```

Comment:
Why do we need the error message in addition to the error code?

Now we use the error message because the error kinds are very broad, so the message helps clarify the sub-type of error:

```rust
Err(SdkError::SendPaymentFailed {
    err: "Invoice already paid".into(),
}),
```

But if we add error codes, the message becomes superfluous, no?

```rust
Err(SdkError::SendPaymentFailed {
    code: SendPaymentFailedCode::AlreadyPaid,
    err: "Invoice already paid".into(),
}),
```

If the message is always meant to stay the same, and we want to include it for logging or display purposes, then perhaps a constructor for every error kind can help:

```rust
impl SdkError {
    pub fn send_payment_failed() -> Self {
        SdkError::SendPaymentFailed {
            code: SendPaymentFailedCode::AlreadyPaid,
            err: "Invoice already paid".into(),
        }
    }
}
```

so we can just call these methods when we have to instantiate an error type:

```rust
Err(SdkError::send_payment_failed())
```

Reply:
The message can be used to add extra context for debugging etc.

```
SdkError::PaymentError {
    code: PaymentErrorCode::DecodingFailed,
    err: format!("Failed to decode hex payment hash: {e}"),
}

SdkError::PaymentError {
    code: PaymentErrorCode::DecodingFailed,
    err: format!("Failed to decode hex payee pubkey: {e}"),
}
```

Good idea with the constructor for repeated errors 👍